### ISSUE TITLE
SMS emulation uses autodetection of Sega 8-bit slot even when inappropriate

### ISSUE THREAD

**Issue by rnjacobs** [mamedev/mame#8018]
Test case:
Construct a bogus SMS ROM consisting of 16384 byte 0xFF, followed by 16384 byte 0x32
Load using
mame smsj -cart bogus.sms
Result: Emulated machine in infinite loop, as expected
Load using
mame sms -cart bogus.sms
Result:
mame fails to load, with the error message
```
Ignoring MAME exception: Unknown slot option 'dahjee_typea' in slot 'slot'
Unknown slot option 'dahjee_typea' in slot 'slot'
```

This is somehow due to the detection in src/machine/sms.cpp : sms_state::setup_media_slots which doesn't check if the emulated machine can support the Dahjee memory expansion (src/bus/sega8/sega8_slot.cpp)

Expected result:
mame sms -cart bogus.sms results in the BIOS rejecting the ROM. I don't have any preference or idea whether the Dahjee memory should work in the "sms" emulated machine, only that it not yield a fatal error.

**Comment by 0kmg** [comment]
Are you doing homebrew development and testing in MAME? As a work around I'd recommend changing your fill value, or even just changing one byte. The autodetect code in sega8_slot.cpp specifically looks for 0xFF filling the entire 2nd 8KB of ROM as part of the sign that it's a Dahjee cart.

**Comment by rnjacobs** [comment]
Yes, I was doing homebrew and testing in MAME. However, advising me to "just change the padding" is worse than useless because by the time I figured out why I tickled MAME's autodetection, I already knew how to fix it in the homebrew.

It should either 
1- give me a useful error message that tells me what I did wrong, or
2- not autodetect the dahjee memory expansion in a context that doing so will cause it to crash.
Either's fine. Crashing is not.

**Comment by 0kmg** [comment]
Sorry, I'm just another guy on the Internets who was trying to be helpful. You can either "change the padding" or add your rom to the software list to set the cart interface type. That's what's available now. Maybe a volunteer developer with free time will come and write free software to your specifications sooner than later.

**Comment by angelosa** [comment]
Bit late to the party but OP is right about user being able to mount an arbitrary slot type, this is not possible atm by not being able to mount anything marked with `device.option_add_internal`. I'm not even sure why this is a thing if you ask me.
